### PR TITLE
Move onion requests to typescript

### PR DESCRIPTION
**Pull request by msgmaxim**
- Moved onion requests to typescript
- Fixed a few type inconsistencies that typescript found

### CODE REVIEW

File: ts/session/onions/index.ts [hunk 1]
```diff
@@ -0,0 +1,310 @@
+import { allowOnlyOneAtATime } from '../../../js/modules/loki_primitives';
+import * as Data from '../../../js/modules/data';
+import * as SnodePool from '../snode_api/snodePool';
+import _ from 'lodash';
+import fetch from 'node-fetch';
+
+type Snode = SnodePool.Snode;
+
+const MIN_GUARD_COUNT = 2;
```
Comment (vincentbavitz) [inline ts/session/onions/index.ts:9]: Should be in `CONSTANTS` no?

Comment (msgmaxim) [reply ts/session/onions/index.ts]: Updated

File: ts/receiver/receiver.ts [hunk 2]
```diff
@@ -325,6 +325,9 @@ export async function handleUnencryptedMessage({ message: outerMessage }: any) {
     // Public chat messages from ourselves should be outgoing
     type: isPublicChatMessage && isOurDevice ? 'sent' : 'message',
     data: outerMessage,
+    confirm: () => {
+      /* do nothing */
+    },
```
Comment (Bilb) [inline ts/receiver/receiver.ts:330]: shouldn't we remove all calls to this function if it's not needed anymore?

Comment (Bilb) [reply ts/receiver/receiver.ts:330]: sometimes confirm, is set so we cannot remove it completely

**Pull request merged**